components/camel-dynamic-router/src/main/docs/dynamic-router-control-component.adoc

```diff
@@ -0,0 +1,247 @@
+real-time. Participants can choose between using URI query parameters, and sending a control message as the exchange
+message body.
+
+== URI format
+
+[source]
+----
+dynamic-router-control:controlAction[?options]
+----
+
+// component-configure options: START
+// component-configure options: END
+
+// component options: START
+include::partial$component-configure-options.adoc[]
+include::partial$component-endpoint-options.adoc[]
+// component options: END
+
+// endpoint options: START
+// endpoint options: END
+
+=== Subscribing
+
+Subscribing can be achieved by using query parameters in the control endpoint URI, or by sending a
+`DynamicRouterControlMessage` to the control endpoint URI.
+
+===== URI examples
+
+.Example Java URI `RouteBuilder` Subscription
+[source,java]
+----
+// Send a subscribe request to the dynamic router that will match every exchange and route messages to the URI: "direct:myDestination"
+from("direct:subscribe").to("dynamic-router-control:subscribe?subscribeChannel=myChannel&subscriptionId=mySubId&destinationUri=direct:myDestination&priority=5&predicate=true&expressionLanguage=simple");
+----
+
+.Example Java URI `ProducerTemplate` Subscription
+[source,java]
+----
+CamelContext context = new DefaultCamelContext();
+context.start();
+ProducerTemplate template = context.createProducerTemplate();
+RouteBuilder.addRoutes(context, rb -> {
+    // Route for subscriber destination
+    rb.from("direct:myDestination")
+            .to("log:dynamicRouterExample?showAll=true&multiline=true");
+    // Route for subscribing
+    rb.from("direct://subscribe")
+            .toD("dynamic-router-control://subscribe" +
+                    "?subscribeChannel=${header.subscribeChannel}" +
+                    "&subscriptionId=${header.subscriptionId}" +
+                    "&destinationUri=${header.destinationUri}" +
+                    "&priority=${header.priority}" +
+                    "&predicateBean=${header.predicateBean}");
+});
+Predicate predicate = PredicateBuilder.constant(true);
+context.getRegistry().bind("predicate", predicate);
+template.sendBodyAndHeaders("direct:subscribe", "",
+        Map.of("subscribeChannel", "test",
+                "subscriptionId", "testSubscription1",
+                "destinationUri", "direct:myDestination",
+                "priority", "1",
+                "predicateBean", "predicate"));
+----
+Above, because the control URI is dynamic, and since a `ProducerTemplate` does not have a built-in way to send to a
+dynamic URI, we have to send subscription parameters from a `ProducerTemplate` in a different way.  The dynamic-aware
+endpoint uses headers "under the hood", because the URI params are converted to headers, so we can set the headers
+deliberately.
+
+.Example XML URI Subscription
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
```

Comment:
We are not recommending old Spring <beans> today. Can you just make the example with Camel <route>

Reply:
I will make sure to convert all of the examples that use spring xml to something else (perhaps Spring Java config).  I can also do that for the integration tests that use old spring xml.

Reply:
No just remove <camelContext> and only show generic <route> then it works anywhere with XML

Reply:
It's removed.